googleapiclient/discovery.py

```diff
@@ -185,22 +190,60 @@ def build(serviceName,
 
   requested_url = uritemplate.expand(discoveryServiceUrl, params)
 
+  content = _retrieve_discovery_doc(requested_url, http, cache_discovery, cache)
+
+  return build_from_document(content, base=discoveryServiceUrl, http=http,
+      developerKey=developerKey, model=model, requestBuilder=requestBuilder,
+      credentials=credentials)
+
+
+def _retrieve_discovery_doc(url, http, cache_discovery, cache=None):
+  """Retrieves the discovery_doc from cache or the internet.
+
+  Args:
+    url: string, the URL of the discovery document.
+    http: httplib2.Http, An instance of httplib2.Http or something that acts
+      like it that HTTP requests will be made through.
+    cache_discovery: Boolean, whether or not to cache the discovery doc.
+    cache: googleapiclient.discovery_cache.base.CacheBase, an optional
+      cache object for the discovery documents.
+
+  Returns:
+    A unicode string representation of the discovery document.
+  """
+  if http is None:
+    http = httplib2.Http()
+
+  if cache_discovery:
+    from . import discovery_cache
+    from .discovery_cache import base
+    if cache is None:
+      cache = discovery_cache.autodetect()
+    if cache:
+      if isinstance(cache, base.CacheBase):
```

Comment:
Typechecking in Python is generally only appropriate when not doing so has the potential to silently corrupt a system and generate symptoms far removed from the ultimate cause of the defect. That doesn't seem to be the case here - the cache is immediately used and its get method is called. Drop the typecheck?